pkgs/top-level/lua-packages.nix

```diff
@@ -270,22 +299,38 @@ let
 
   luaposix = buildLuaPackage rec {
     name = "posix-${version}";
-    version = "33.4.0";
+    version = "34.0.4";
 
     src = fetchFromGitHub {
       owner = "luaposix";
       repo = "luaposix";
       rev = "release-v${version}";
-      sha256 = "0y531p54lx2yf243bcsyp6sv8fvbqidp20yry0xvb85p8zw9dlrq";
+      sha256 = "0p5583vidsm7s97zihf47c34vscwgbl86axrnj44j328v45kxb2z";
     };
 
-    buildInputs = [ perl ];
+    propagatedBuildInputs = [ std.normalize bit32 ];
```

Comment:
Is `bit32` needed for lua5.2?

Reply:
Seems like luaposix library doesn't check if running under lua5.2, it loads bit32 lib every time. I would leave it.

Reply:
ok.